src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,213 @@
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import React, { useEffect, useState, useRef } from 'react';
+import { patchTodo } from '../../api/todos';
+import { deleteTodos } from '../../api/todos';
+import { ErrorMessage } from '../../types/ErrorMessage';
+
+type Props = {
+  todo: Todo;
+  onDeleteTodo: (id: number) => void;
+  todoIdsInLoading: number[];
+  onUpdateTodo: (updatedTodo: Todo) => void;
+  onSetTodoIdsInLoading: React.Dispatch<React.SetStateAction<number[]>>;
+  onSetTodo: React.Dispatch<React.SetStateAction<Todo[]>>;
+  onHandleErrorShow: (_: string) => void;
+  todoEditingId: number | null;
+  onSetTodoEditingId: React.Dispatch<React.SetStateAction<number | null>>;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDeleteTodo,
+  onUpdateTodo,
+  todoIdsInLoading,
+  onHandleErrorShow,
+  onSetTodoIdsInLoading,
+  onSetTodo,
+  todoEditingId,
+  onSetTodoEditingId,
+}) => {
+  const [isTitleEditing, setIsTitleEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+  const [isHasError, setIsHasError] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isTitleEditing, isHasError]);
+
+  useEffect(() => {
+    setIsTitleEditing(isHasError);
+  }, [isHasError]);
+
+  useEffect(() => {
+    const handleKeyUp = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        setIsTitleEditing(false);
+        document.removeEventListener('keyup', handleKeyUp);
```

Comment:
Can we move this logic to `onKeyUp` react event?

https://www.geeksforgeeks.org/react-onkeyup-event/